lib/Listener/LoadViewerListener.php

```diff
@@ -56,6 +63,7 @@ public function handle(Event $event): void {
 		}
 		if ($this->permissionManager->isEnabledForUser() && $this->userId !== null) {
 			$this->initialStateService->provideCapabilities();
+			$this->eventDispatcher->dispatchTyped(new RenderReferenceEvent());
```

Comment:
document.js is loaded in an iframe already, so triggering the event rather ehe should work: https://github.com/nextcloud/richdocuments/blob/e3105a17e6b9f4baa4182f97d05f7b6f39e4d7a9/lib/Controller/DocumentTrait.php#L15

This is how the frames are currently structured, I've been thinking for a while if we shouldn't refactor that at some point but that may be larger work:

- Nextcloud loads viewer.js
  - The Office.vue component loads an iframe rendering a simple template through the DocumentController with document.js
    - document.js submits the wopi data to Collabora within another iframe

Reply:
:heart: